fahrschein/src/main/java/org/zalando/fahrschein/NakadiClient.java

```diff
@@ -59,8 +61,19 @@ public static NakadiClientBuilder builder(URI baseUri, RequestFactory requestFac
         this.objectMapper = objectMapper;
         this.internalObjectMapper = DefaultObjectMapper.INSTANCE;
         this.cursorManager = cursorManager;
+        this.eventPublishingHandlers = new ArrayList<>();
```

Comment:
If we persist a reversed version of `eventPublishingHandlers`, we can do callbacks after the request in the reverse order, preserving visibility of the effects of each handler.